src/components/form.tsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import './form.css';
+import { IFormValues, NotValidInputs } from './formTypes';
+import LabledInput from './labledInput';
+import validateForm from './helpers';
+import LabledSelect from './labledSelect';
+import LabledSwitcher from './labledSwitcher';
+import { IPersonCardProps } from './personCard';
+
+interface IFormState {
+  isSubmitDisabled: boolean;
+  isFirstFilling: boolean;
+  isNotValid: NotValidInputs;
+}
+
+interface IFormProps {
+  addPersonCard: (personCard: IPersonCardProps) => void;
+}
+
+class Form extends React.Component<IFormProps, IFormState> {
+  errMsg: IFormValues = {
+    name: 'should not be empty',
+    surname: 'should not be empty',
+    birthday: 'select more than 11yo',
+    picture: 'please select picture',
+    consent: 'please check to proceed',
+    country: 'please select country',
+    gender: 'please select your gender',
+  };
+
+  notValidInputsCount: number;
+
+  countries = ['Italy', 'Norway', 'Germany', 'Spain', 'Sweden', 'Ukraine', 'USA'];
+  genders = ['male', 'female'];
+  formRef: React.RefObject<HTMLFormElement>;
+
+  constructor(props: IFormProps) {
+    super(props);
+    this.state = {
+      isSubmitDisabled: true,
+      isFirstFilling: true,
+      isNotValid: {
+        name: false,
+        surname: false,
+        birthday: false,
+        picture: false,
+        consent: false,
+        country: false,
+        gender: false,
+      },
+    };
+    this.notValidInputsCount = 0;
```

Comment:
In react we do not store state like this, if you need something that you want to store between submits then try to use state of the component or in some rare cases ref